src/core/reference/src/runtime/reference/utils/fft_common.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "ngraph/runtime/reference/utils/fft_common.hpp"
+
+#include <algorithm>
+#include <cstddef>
+#include <cstring>
+#include <numeric>
+
+#include "ngraph/check.hpp"
+
+namespace ngraph {
+namespace runtime {
+namespace reference {
+namespace fft_common {
+std::vector<int64_t> reverse_shape(const ngraph::Shape& shape) {
+    size_t complex_data_rank = shape.size() - 1;
+
+    std::vector<int64_t> reversed_shape(complex_data_rank);
+    for (size_t i = 0; i < complex_data_rank; ++i) {
+        reversed_shape[i] = static_cast<int64_t>(shape[complex_data_rank - i - 1]);
+    }
+    return reversed_shape;
+}
+
+std::vector<int64_t> compute_strides(const std::vector<int64_t>& v) {
+    std::vector<int64_t> strides(v.size() + 1);
+    int64_t stride = 1;
+    for (size_t i = 0; i < v.size(); ++i) {
+        strides[i] = stride;
+        stride *= v[i];
+    }
+    strides.back() = stride;
```

Comment:
Why is strides vector size bigger than vector?

Reply:
Because if a vector is `[v_0, v_1, ..., v_{n-1}]`, then the strides vector is `[1, v_0, v_0v_1, v_0v_1v_2, ..., v_0v_1...v_{n-1}]`. And the last element, `v_0v_1v_2, ..., v_0v_1...v_{n-1}` is used as number of elements in the tensor of the shape  `[v_0, v_1, ..., v_{n-1}]`.